src/main/java/cn/nukkit/item/Item.java

```diff
@@ -310,7 +304,9 @@ public static void init() {
         initCreativeItems();
     }
 
-    private static final ArrayList<Item> creative = new ArrayList<>();
+    public static void registerCustomItemBlock(int key, Class i) {
```

Comment:
this whole method needs to be made better. at the very least a `null` check should be added, but it should also check for ID conflicts with itemblocks that are already registered. maybe some kind of automatic ID allocation system (similar to the way Minecraft Forge does it with resource locations as registry keys) would be in order.

Reply:
Well Imo, I think that only 1 Custom Item block class should be allowed per block just because, each class handled the meta and trying to figure out which plugin should override the other,.. Maybe In a later update force the Plugins to declare the Blocks That they want to register in their Config.yml then on Plugin Load it will load the Allowed CustomItemBlocks